### PR TITLE
Update go.mod dependencies to address vulnerabilities

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

We need to update a few dependencies to ensure that we are able to remove from the go.mod the github.com/hashicorp/vault >= 1.10.0, < 1.10.3 to address vulnerabilities.

Screenshot 2022-11-23 at 12 28 36

## Steps to reproduce

Check https://github.com/replicatedhq/kURL/security/dependabot

#### Does this PR introduce a user-facing change?

NONE
#### Does this PR require documentation?
NONE

### CODE REVIEW

**Review by laverya — APPROVED** [review]

**Review by emosbaugh — CHANGES_REQUESTED** [review]
Can you hold off on this one until i get my rook pr merged? https://github.com/replicatedhq/ekco/pull/84

**Comment by camilamacedo86** [comment]
Hi @emosbaugh,

I reverted the change to upgrade the rock (so we can achieve the goal without be blocked by). So could we move with this one?

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**
